accession-commons-monotonic-generator-jpa/src/test/java/uk/ac/ebi/ampt2d/commons/accession/persistence/jpa/monotonic/service/ContiguousIdBlockServiceTest.java

```diff
@@ -130,8 +130,97 @@ public void testBlockSizeAndIntervalWithMultipleInstances() {
         assertEquals(0, block1.getFirstValue());
         assertEquals(999, block1.getLastValue());
         ContiguousIdBlock block2 = service.reserveNewBlock(CATEGORY_ID_2, INSTANCE_ID_2);
-        assertEquals(3000, block2.getFirstValue());
-        assertEquals(3999, block2.getLastValue());
+        assertEquals(2000, block2.getFirstValue());
+        assertEquals(2999, block2.getLastValue());
         assertEquals(block2, repository.findFirstByCategoryIdOrderByLastValueDesc(CATEGORY_ID_2));
+
+        //Manually save a block of size 500, so for the current range only a block size of 500 reserved
+        repository.save(new ContiguousIdBlock(CATEGORY_ID_2, INSTANCE_ID, 4000, 500));
+        //Reserve a new block with size 1000
+        ContiguousIdBlock block3 = service.reserveNewBlock(CATEGORY_ID_2, INSTANCE_ID_2);
+
+        assertEquals(4500, block3.getFirstValue());
+        // the block is reserved only with size 500 the remainingSize in the range is 500 only
```

Comment:
Clarify this: the block was reserved with size 1000, but only 500 were available due to the interleaving.